### DISCUSSION TITLE
problem with useSound, vercel deployment issue.

### DISCUSSION THREAD

**Discussion by PhilipZi** [neuefische/questions#126] [Web] [answered]
Hallo zusammen, 

ich habe in meinem Capstone folgendes Problem, bei meinem aktuellstem commit kommt es zu einem deployment Fehler. Siehe Bild.
![Screenshot 2022-08-30 at 11 47 41](https://user-images.githubusercontent.com/102238180/187406061-d34f0ca2-f289-4f3d-ba74-5e83355deef8.png)

ich importiere wie in der Docu beschrieben useSound wie folgt (siehe Bild) 
![Screenshot 2022-08-30 at 11 48 44](https://user-images.githubusercontent.com/102238180/187406286-6ef9a272-c162-46f0-b6b4-c5dcf5610962.png)

hier der Link zu der Libary: 
https://www.joshwcomeau.com/react/announcing-use-sound-react-hook/

hier der Link zu meinem github pullrequest: 
https://github.com/PhilipZi/capstone-project/pull/22

**Comment by alexclaes** [comment, marked as answer]
Hallo Philip,

das liegt in der Regel daran, dass die Dependency nicht korrekt installiert ist (also nicht in der `package.json` steht).

Habe mir den branch mal auf meinen Rechner gezogen und einn `npm run build` ausprobiert (im Grunde macht Vercel dasselbe).
Bei mir lokal funktioniert das. In der `package.json` ist auch alles richtig eingetragen.

Laut den Infos in dem PR ist das Deployment auch richtig durchgelaufen. Besteht das Problem noch?

**Reply by PhilipZi** [reply]
Ja genau, danke Alex, ich wollte jetzt nochmal ausführlich schreiben was das problem war und was man noch machen sollte, dass useSound einwandfrei läuft. Danke für das rüber schauen. Jessica hat mir da sehr gut helfen können. 
Ich hatte auf meinem prototype branch das package installiert aber dann nicht auf meinem aktuellem feature branch, das war das haupt Problem.

**Comment by PhilipZi** [comment]
Wie Alex oben schon beschrieben hat, musste ich nochmal auf dem richtigen Branch "npm install use-sound" ausführen, damit das Vercel probelm behoben werden konnte. 

AUßERDEM: 
empfehle ich eine next.config.js Datei anzulegen auf der obersten Ordnerstruktur, in welche folgende config Daten rein müssen, dass in der Console keine Warnungen oder Fehler angezeigt werden und useSound einwandfrei funktioniert. Hier der Code: 

```
module.exports = {
	webpack(config, options) {
		const {isServer} = options;
		config.module.rules.push({
			test: /\.(ogg|mp3|wav|mpe?g)$/i,
			exclude: config.exclude,
			use: [
				{
					loader: require.resolve('url-loader'),
					options: {
						limit: config.inlineImageLimit,
						fallback: require.resolve('file-loader'),
						publicPath: `${config.assetPrefix}/_next/static/images/`,
						outputPath: `${isServer ? '../' : ''}static/images/`,
						name: '[name]-[hash].[ext]',
						esModule: config.esModule || false,
					},
				},
			],
		});

		return config;
	},
};
````
hier nochmal ein Bild dazu: 
![Screenshot 2022-08-30 at 12 46 38](https://user-images.githubusercontent.com/102238180/187417678-4c03b51b-5736-47fc-bcdd-3b5b050f3888.png)
